src/components/AccountManager/AccountManager.js

```diff
@@ -0,0 +1,253 @@
+        }
+        setResetPasswordOpen(!isResetPasswordOpen);
+    };
+
+    const [isFaceSetupOpen, setFaceSetupOpen] = useState(false);
+    const toggleFaceSetup = () => setFaceSetupOpen(!isFaceSetupOpen);
+
+    const [isGoogleOpen, setIsGoogleOpen] = useState(false);
+    const toggleGoogle = () => setIsGoogleOpen(!isGoogleOpen);
+
+    const context = useContext(AppContext);
+    const userInfo = useDatabase();
+
+    useEffect(() => {
+        const getUserInfo = async () => {
+            let info = null;
+            await userInfo.findOne('users', {userId: context.mongoHook.authenticatedUser.id})
+                .then(authedUserInfo => {
+                    info = authedUserInfo;
+                });
+
+            if(info !== null){
+                setEmail(info.email);
+                setFirstName(info.first_name);
+                setLastName(info.last_name);
+
+                if(info.guest === "false" || !info.guest){
+                    setIsGuest(false);
+                }
+            }
+            else{
+                console.log("why null?")
+            }
+
+            console.log("Guest? " + isGuest.toString());
+        };
+
+        // const getInfo = async () => {
+        //     if(!isGuest){
+        //         let info = await userInfo.findOne('face_descriptors', {owner_id: context.mongoHook.authenticatedUser.id})
+        //             .then(authedUserInfo => {
+        //                 return authedUserInfo;
+        //             });
+        //         if (info != null) {
+        //             setPassword(info.password); //DEV USE ONLY - DELETE AFTER TESTING
+        //         }
+        //     }
+        // };
+
+        //getInfo();
+        getUserInfo();
+
+        if(resetPasswordOne === "" || resetPasswordTwo === ""){
+            setPasswordsMatch("Enter your new password...");
+        }
+        else if(resetPasswordOne === resetPasswordTwo){
+            setPasswordsMatch("Passwords Match!");
+        }
+        else {setPasswordsMatch("Passwords do not match!")}
+
+    }, [updateInfo, resetPasswordOne, resetPasswordTwo]);
+
+    useEffect(() => {
+        if(!isGuest){
+            console.log("NotAGuest"+isGuest.toString());
+            setConfirmationText("Confirm");
+        }
+        else{
+            console.log("IsAGuest"+isGuest.toString());
+            setConfirmationText("DISABLED");
+        }
+    }, [isGuest]);
+
+    const changeInfo = async () => {
+        if(!isGuest){
+            let msg = await userInfo.updateOne("users", {userId: context.mongoHook.authenticatedUser.id},
+                {$set: {email: changeEmail, first_name: changeFirstName, last_name: changeLastName}});
+            setUpdateInfo(!updateInfo);
+            switch(msg.matchedCount){
+                case 0: msg = "Matched: 0\nAccount not found!";
+                    break;
+                case 1: msg = "Matched: 1\nAccount found!\n"+context.mongoHook.authenticatedUser.id;
```

Comment:
You may want to add a null check to `context.mongoHook.authenticateduser.id`. You can be fairly certain that `context.mongoHook` isn't null, but the `.authenticatedUser` and definitely the `.authenticatedUser.id` calls could possibly crash

Reply:
Changed